engine/src/main/java/org/terasology/engine/modes/StateLoading.java

```diff
@@ -243,8 +232,16 @@ public void update(float delta) {
         EngineTime time = (EngineTime) context.get(Time.class);
         long startTime = time.getRealTimeInMs();
         while (current != null && time.getRealTimeInMs() - startTime < 20 && !gameEngine.hasPendingState()) {
-            if (current.step()) {
-                popStep();
+            try {
+                if (current.step()) {
+                    popStep();
+                }
+            } catch (Exception e) {
```

Comment:
Do we have more concrete information about the errors that can appear here? Catching the general `Exception` class is kind of broad, we might want to refine that if possible.

Reply:
Being broad is kinda the point of #4004!  Oh, I failed to mention that this PR is based on that one, didn't I. Oops. Anyway, I think that's where this thread should happen.